### PR TITLE
fix SDL2 detection for Visual

### PR DESCRIPTION
**Pull request by Lectem**
This fixes the default installation hierarchy of SDL2, so that users only have to extract and set SDL2DIR.
This was inspired by the following official cmake patch for FindSDL.cmake https://cmake.org/gitweb?p=cmake.git;a=commitdiff;h=96197198

### CODE REVIEW

**Comment by MerryMage** [comment]
I hadn't considered Windows users not using the bundled libraries.

LGTM

File: externals/cmake-modules/FindSDL2.cmake [hunk 1]
```diff
@@ -134,11 +134,17 @@ SET(SDL2_SEARCH_PATHS
...
+else()
+    set(VC_LIB_PATH_SUFFIX lib/x86)
+endif()
+
 FIND_LIBRARY(SDL2_LIBRARY_TEMP
     NAMES SDL2
     HINTS
     $ENV{SDL2DIR}
-    PATH_SUFFIXES lib64 lib
+    PATH_SUFFIXES lib64 lib  ${VC_LIB_PATH_SUFFIX}
```
Comment (MerryMage) [inline externals/cmake-modules/FindSDL2.cmake]: Is the double space necessary?

Comment (Lectem) [inline externals/cmake-modules/FindSDL2.cmake]: nope, it's a typo, I'll update the PR

**Comment by bunnei** [comment]
LGTM

**Pull request merged**